scripts/check_docs_formatted.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+"""Check code snippets in docs are formatted by black."""
+import argparse
+import os
+import re
+import textwrap
+from collections.abc import Sequence
+from pathlib import Path
+from re import Match
+
+import black
+from black.mode import Mode, TargetVersion
+from black.parsing import InvalidInput
+
+TARGET_VERSIONS = ["py37", "py38", "py39", "py310", "py311"]
+SNIPPED_RE = re.compile(
+    r"(?P<before>^(?P<indent> *)```\s*python\n)"
+    r"(?P<code>.*?)"
+    r"(?P<after>^(?P=indent)```\s*$)",
+    re.DOTALL | re.MULTILINE,
+)
+
+# For some rules, we don't want black to fix the formatting as this would "fix" the
+# example.
+KNOWN_FORMATTING_VIOLATIONS = [
```

Comment:
I opted to just inline these, it felt a little simpler than maintaining separate files that would only be used here, but let me know if you disagree.